src/lingodotdev/engine.py

```
@@ -18,16 +15,24 @@ class EngineConfig(BaseModel):
-    api_url: str = "https://engine.lingo.dev"
+    engine_id: str
+    api_url: str = "https://api.lingo.dev"
     batch_size: int = Field(default=25, ge=1, le=250)
     ideal_batch_item_size: int = Field(default=250, ge=1, le=2500)
 
-    @validator("api_url")
+    @validator("engine_id", pre=True, always=True)
+    @classmethod
+    def validate_engine_id(cls, v: str) -> str:
+        if not v.strip():
+            raise ValueError("engine_id is required and cannot be empty")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)